### ISSUE TITLE
CA2000 not triggered for HttpResponseMessage

### ISSUE THREAD

**Issue by LeeGrissom** [dotnet/roslyn-analyzers#5794]
### Analyzer

**Diagnostic ID**: [CA2000](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2000): `Dispose objects before losing scope`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 6 SDK](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 6.0.101](https://dotnet.microsoft.com/en-us/download/dotnet/6.0)

### Describe the bug

I Set severity on rule CA2000 to Warning.
After running Code Analysis on solution I was expecting the following C# .NET 6 console app to trigger CA2000 rule on line 3 `HttpResponseMessage`.
However, Code Analysis results show no warning message indicating that there is no memory leak.
**Why?**
```c#
const string uri = "https://tools.learningcontainer.com/sample-json.json";
using HttpClient client = new();
HttpResponseMessage response = await client.GetAsync(uri);
var responseBody = await response.Content.ReadAsStringAsync();
Console.WriteLine(responseBody);

```

### Steps To Reproduce

1. Launch Visual Studio 2022.
2. Create new console project using .NET 6.0.101
3. Paste the code above into the `Program.cs` file
4. In solution explorer, drill down into the CA2000 rule and set severity to warning (Dependencies -> Analyzers -> Microsoft.CodeAnalysis.NetAnalyzers -> CA2000)
5. Click `Analyze` main menu, then click `Run code analysis -> on solution` menu item
6. Open the `Error List` tool window and ensure you're viewing `Build + Intellisense` output, and ensure you have the `Warnings` button engaged.
7. Notice that no warning is displayed for CA2000.

### Expected behavior
When CA2000 is enabled, a violation message should be emitted. 

### Actual behavior
No violation message is emitted.

**Comment by jvilimek** [comment]
Same here for abstract methods:
```C#
BaseTest t = new Test();
t.M(); // this line should trigger the CA2000 rule, but WON'T
((Test)t).M2(); //this line triggers the CA2000 rule

class A : IDisposable
{
    public void Dispose()
    {
    }
}

abstract class BaseTest
{
    public abstract A M();
}
class Test : BaseTest
{
    public override A M()
    {
        return new A();
    }
    public A M2()
    {
        return new A();
    }
}
```
